libs/cardano-data/cardano-data.cabal

```diff
@@ -20,7 +20,6 @@ library
         Data.CanonicalMaps
         Data.Pulse
         Data.MapExtras
-        Data.UMap
```

Comment:
Module has been removed. That is a breaking change, therefore it needs to be reflected in a cabal version and the changelog. 

Moreover, this removal affects older versions of `set-algebra`, so if you feel adventurous you could [add a revision](https://github.com/input-output-hk/cardano-haskell-packages#how-to-add-a-new-package-metadata-revision) on the already released [version of `set-algebra` ](https://github.com/input-output-hk/cardano-haskell-packages/blob/main/_sources/set-algebra/1.0.0.0/meta.toml) and add an upper bound on the `cardano-data`, but only after this PR is merged, of course.